swedbank-pay-woocommerce-payments.php

```diff
@@ -146,376 +68,22 @@ public function plugin_action_links( $links ) {
-			exit( 'No naughty business' );
-		}
-
-		$order_id = (int) $_REQUEST['order_id'];
-
-		try {
-			swedbank_pay_capture_payment( $order_id );
-			wp_send_json_success( __( 'Capture success.', WC_Swedbank_Pay::TEXT_DOMAIN ) );
-		} catch ( Exception $e ) {
-			$message = $e->getMessage();
-			wp_send_json_error( $message );
-		}
-	}
-
-	/**
-	 * Action for Cancel
-	 */
-	public function ajax_swedbank_pay_cancel() {
-		if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'swedbank_pay' ) ) {
-			exit( 'No naughty business' );
-		}
-
-		$order_id = (int) $_REQUEST['order_id'];
-
-		try {
-			swedbank_pay_cancel_payment( $order_id );
-			wp_send_json_success( __( 'Cancel success.', WC_Swedbank_Pay::TEXT_DOMAIN ) );
-		} catch ( Exception $e ) {
-			$message = $e->getMessage();
-			wp_send_json_error( $message );
-		}
-	}
-
-	/**
-	 * Generate UUID
-	 *
-	 * @param $node
-	 *
-	 * @return string
-	 */
-	public function generate_uuid( $node ) {
-		return \Ramsey\Uuid\Uuid::uuid5( \Ramsey\Uuid\Uuid::NAMESPACE_OID, $node )->toString();
-	}
-
-	/**
-     * Get Payment Description
-     *
-	 * @param string $description
-	 * @param WC_Order $order
-	 *
-	 * @return string
-	 */
-	public function payment_description( $description, $order ) {
-		if ( ! $order instanceof WC_Order ) {
-			$order = wc_get_order( $order );
-		}
-
-	    return sprintf( __( 'Order #%s', WC_Swedbank_Pay::TEXT_DOMAIN ), $order->get_order_number() );
-    }
-
-	/**
-	 * Dispatch Background Process
-	 */
-	public function maybe_process_queue() {
-		self::$background_process->dispatch();
-	}
-
-	/**
-	 * Provide Admin Menu items
-	 */
-	public function admin_menu() {
-		// Add Upgrade Page
-		global $_registered_pages;
-
-		$hookname = get_plugin_page_hookname( 'wc-payex-psp-upgrade', '' );
-		if ( ! empty( $hookname ) ) {
-			add_action( $hookname, __CLASS__ . '::upgrade_page' );
-		}
-
-		$_registered_pages[ $hookname ] = true;
-	}
-
-
-	/**
-	 * Upgrade Page
-	 */
-	public static function upgrade_page() {
-		if ( ! current_user_can( 'update_plugins' ) ) {
-			return;
-		}
-
-		// Run Database Update
-		include_once( dirname( __FILE__ ) . '/includes/class-wc-swedbank-pay-update.php' );
-		WC_Swedbank_Pay_Update::update();
-
-		echo esc_html__( 'Upgrade finished.', WC_Swedbank_Pay::TEXT_DOMAIN );
-	}
-
-	/**
-	 * Upgrade Notice
-	 */
-	public static function upgrade_notice() {
-		if ( current_user_can( 'update_plugins' ) ) {
-			?>
-			<div id="message" class="error">
-				<p>
-					<?php
-					echo esc_html__( 'Warning! Swedbank Pay WooCommerce payments plugin requires to update the database structure.', WC_Swedbank_Pay::TEXT_DOMAIN );
-					echo ' ' . sprintf( esc_html__( 'Please click %s here %s to start upgrade.', WC_Swedbank_Pay::TEXT_DOMAIN ), '<a href="' . esc_url( admin_url( 'admin.php?page=wc-payex-psp-upgrade' ) ) . '">', '</a>' );
-					?>
-				</p>
-			</div>
-			<?php
-		}
+		// Register Gateways
+		WC_Swedbank_Pay::register_gateway( 'WC_Gateway_Swedbank_Pay_Cc' );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class 'WC_Swedbank_Pay' in method 'woocommerce_loaded'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)